### ISSUE TITLE
Very laggy on Android mobile app when scrolling on default photos view

### ISSUE THREAD

**Issue by maltokyo** [immich-app/immich#12674]
### The bug

When scrolling on Google photos, the scrolling is extremely smooth even when scrolling too fast for full thumbnails to load. On Immich this is very laggy. Granted I have 100k photos, so not sure if that's the issue but it should still work. Attached a screen recording of me scrolling at a constant speed, and the laggy scrolling is clearly visible. 

Link to video 
https://youtube.com/shorts/D0theLCMa4s?si=ZsPg36C8EhGr6ktO

### The OS that Immich Server is running on

android 14

### Version of Immich Server

v1.115

### Version of Immich Mobile App

v1.114

### Platform with the issue

- [ ] Server
- [ ] Web
- [X] Mobile

### Your docker-compose.yml content

```YAML
Stock standard docker compose. With Intel HW acceleration on.
```

### Your .env content

```Shell
Standard env
```

### Reproduction steps

1. Scroll on photos screen through many photos at medium pace (constant pace)
2. Observe lagging 
...

**Comment by bo0tzz** [comment]
What device are you using? Try enabling "prefer remote images" in the advanced settings.

**Comment by Cookie-Monster-Coder** [comment]
> What device are you using? Try enabling "prefer remote images" in the advanced settings.

This does not work.

Related issue:
https://github.com/immich-app/immich/issues/11113

**Comment by maltokyo** [comment]
Google pixel 8 pro. All my images are remote anyway. You can see twy only
have the cloud icon on them.

**Comment by maltokyo** [comment]
Anyway have turned on prefer remote images setting. It's still the same.
Very laggy and slow to scroll.

**Comment by alextran1502** [comment]
@maltokyo how many albums do you have? Did you start scrolling right away when you first open the app? 

The stuttering is due to the albums sync mechanism compete with the UI thread, after the albums sync finished, it will be smoother.

Regardless, we have plan to optimize this process for a better user experience

**Comment by maltokyo** [comment]
Thank you for the replies. Yes I used a script to make albums from about 1000 folders. So, many albums. But immich has been installed for weeks. So I thought it maybe has had enough time to scroll correctly. Thanks for the news on optimizing this.

**Comment by alextran1502** [comment]
@maltokyo the sync here check for the diff with the server, adding and removing assets on the mobile...etc, basically expensive operations that can be taxing for the client, and it runs when opening the app and after some certain actions.

**Comment by maltokyo** [comment]
Thank you. Everything is in sync. I haven't added any photos for many
weeks. Nothing is there to sync.

**Comment by alextran1502** [comment]
@maltokyo maybe a better wording from is the operation of figuring out what to sync is taxing and competing with UI thread

**Comment by maltokyo** [comment]
@alextran1502 ok, is there anything I can do to get more data on this on my side, in order to find out what is causing the issue? Some way to get logs (and which ones?) from mobile or server, please let me know. Any way I can assist, I'm willing to of course.

**Comment by alextran1502** [comment]
@maltokyo thanks for the offer, we already known the bottle neck and have plans to word toward improving it in the near future

**Comment by maltokyo** [comment]
Understood. Thanks a lot. Apart from this, everything is absolutely amazing.

**Comment by Mitmischer** [comment]
I'd also like to help debug this issue. For me the experience is even worse - the app takes a long time to load and becomes completely frozen for multiple seconds sometimes, most often directly after starting while synchronization takes place. Having "remove images preferred" activated at this point doesn't help, but it helps later during normal use. Maybe it's also my device but it should be that out-of-date and slow yet (Snapdragon 845, no SD card).

Thanks for the hard work, for the rest I can only agree - Immich is amazing!

**Comment by sbillaudelle** [comment]
I experience the same issue. So far, I had attributed this to the age of my phone, but I just upgraded and still suffer from a laggy UI.

~ 20 seconds after opening the app, it becomes very fluid and responsive, corroborating the statement that this is connected to the synchronization.

Please let me know if you need more data.

**Comment by orywig** [comment]
I experience the same thing on iOS. Did not have this issue until I imported a Google Takeout which added a ton of albums. Which, I only have 133 albums, which doesn’t sound like a ton.

**Comment by alextran1502** [comment]
@orywig yeah the album sync is causing the competition with the UI thread at the moment. The looping over albums and each assets to calculate which one need to be add/delete/update is quite taxing, so we are working towards a fix

**Comment by orywig** [comment]
Thanks, @alextran1502! I should have mentioned I saw your comments above about there being a plan for this. I just wanted to note that it’s a iOS thing as well as Android. Thanks for all you do! Love the project!

**Comment by orywig** [comment]
Until the team can resolve this issue, has anyone here found a temporary resolution or workaround? The lag while the albums are syncing is quite impactful to the user experience; especially when remote and this is happening across the internet. I can't think of anything besides just cleaning up and removing any unnecessary albums to shorten the sync time.

**Comment by Cookie-Monster-Coder** [comment]
I don't believe the full cause of this issue is albums syncing. I open up the app, wait 5 mins on the startup position, just taping the screen every so often so the phone doesn't go to sleep, after the 5 mins are up I start scrolling and it still crashes.

**Comment by alextran1502** [comment]
@Cookie-Monster-Coder hmmm, what is your phone model? If you go to App Settings > Advanced > toggle Prefer Remote Image, does it help?

**Comment by maltokyo** [comment]
I also have the same, open app, wait 20-30 mins even..  and lags and
crashes still. I'm on pixel 8 pro.

**Comment by maltokyo** [comment]
Describes my situation perfectly Jörg

**Comment by Cookie-Monster-Coder** [comment]
> @Cookie-Monster-Coder hmmm, what is your phone model? If you go to App Settings > Advanced > toggle Prefer Remote Image, does it help?

Samsung 22+

Enabling Remote Images does NOT help.

**Comment by alextran1502** [comment]
Hmm, this is very odd; I tested my 80_000 assets library on a Pixel 8 and Galaxy 23 and cannot cause it to crash or stutter after the assets and album are synced. I wonder if there is some specific photo that the app cannot read and cause the crash 🤔

**Comment by maltokyo** [comment]
Could it be that I'm using an external script to create albums from each
folder in the external library? That has created 1000s of albums.

**Comment by alextran1502** [comment]
@maltokyo Yep, that will do it for your case. Until the new album sync is implemented, it is not pretty for 1000s albums at the moment

**Comment by alextran1502** [comment]
@maltokyo The current album sync process is running on the main thread due to its complexity and causing the UI to stutter

**Comment by maltokyo** [comment]
Ok, thanks for the clarification. Makes sense. If somehow I can delete all
albums and try without them, happy to give it a try. Just not sure how to
delete 1000s albums.

**Comment by ratti** [comment]
I can confirm I have 3022 albums. My external folder is just a rsync from my Mac, and there the photos are organized by a schema
{$year}  /  {$year_$month_$day___Some_Nice_Description}

**Comment by ldandersen** [comment]
I have a library of 50000+ photos (with only a few albums and basically a pretty stock Immich server installation), and can confirm that for me the app has become borderline unusable on iOS. Web interface works like a champ on both mobile and desktop.

Disabling sync does seem to mitigate the problem, which seems to point (as some above have speculated) to the issue being the sync process blocking the main thread--for seemingly up to 10-20 seconds at a time on a library as large as mine. If this is true, it's definitely not a sustainable way to approach the sync process.

Anyway, pretty bummed because this is the first real issue I've had with what is otherwise an absolutely stellar project!

**Comment by alextran1502** [comment]
@ldandersen If you have a few albums then the app should be smooth after the hashing progress is done (check for the spinner icon on the Photos page)

**Comment by ratti** [comment]
Just to add feedback: Me, huge collection, many albums, doesn't use sync. 

I am syncing the photos from my smartphone with a different app to a different place, then I tag & sort & fix them manually on my computer, then I rsync the fine result to my webserver, and that folder then is mounted into immich as an external dir. 

So, I neither upload manually via app/web, nor I use syncing in the app. Just a mounted folder that's changing ~once a month.

(and: I totally agree this project is shocking great!)

**Comment by mosalski** [comment]
The same issue for me. I have ~100 albums, both from external&internal directory. Scrolling is always quite laggy.

**Comment by ratti** [comment]
Question to those having many albums: Is your "Map" feature broken also?
For me it used to work on >100000 photos, until I created albums. Now the page hangs on opening. Again, only app. Browser is fine.

**Comment by mosalski** [comment]
> Question to those having many albums: Is your "Map" feature broken also? For me it used to work on >100000 photos, until I created albums. Now the page hangs on opening. Again, only app. Browser is fine.

Yes, map is also unusable laggy.

**Comment by alextran1502** [comment]
@tobiasgraeber we don't do ETA but this is high on the list. Unfortunately there are many pieces we have to untangle first before we can make this process better

**Comment by waclaw66** [comment]
It helped to me to clear app data and let app sync everything from scratch. It seems the app synces albums again and again even that there was not change for months/years on them. Couldn't be some db migration issue?

**Comment by Tronix117** [comment]
My wife's phone is a Galaxy S20+ with around 35k assets. We've also got a few other family members (all on Samsung devices) running into the same issue: freezes at startup whenever sync is enabled.

I’ve tried the usual steps: clearing data, uninstalling/reinstalling the app… but the moment sync kicks in, it only runs for a few seconds before freezing during the asset check. The app stays unresponsive for about 1–2 minutes before it recovers.

We're all on the latest version (1.131), and right now this is a major blocker for wider adoption in my circle.

Happy to help troubleshoot, just ping me here or on Discord if you want me to test things or send over some debug info.

**Comment by ldandersen** [comment]
Yeah, I've essentially given up using the syncing functionality on my phone. I just export recent from Apple Photos periodically and drop them into Immich manually.

**Comment by maltokyo** [comment]
Unfortunately, none of my "users" use immich any more, due to the poor performance (with many albums) on the mobile app.  They stick with their google photos / icloud photos etc..  which is sad for me to see.

@alextran1502 is there any help we can give, testing something new, or otherwise.  It appears this could be a pretty big hurdle for user adoption here.

**Comment by alextran1502** [comment]
@maltokyo there is no need, we are rewriting the whole sync mechanism from the ground up, so it will resolve these type of issues

**Comment by ChrislyBear-GH** [comment]
Is there currently any setting/workaround? I guess deleting all albums, but that's not an option for me.

Chris

**Comment by maltokyo** [comment]
No workaround detected by me. For now my users have decided to stay on
Google photos....

**Comment by pwoolvett** [comment]
>The current album sync process is running on the main thread due to its complexity and causing the UI to stutter

My current workaround while we wait for the rewrite is to use the web app for the UI and the actual app in the background only.
By accessing the URL in chrome on mobile and install that, I now have two immich "apps":  the real one which does the work and gets laggy, and the web thing for the ui.

**Comment by akay64** [comment]
Hi! Same issue (1.132.3), S23+ (SD 8 Gen 2, 8GB RAM) , 750 albums, 24000 photos, 1000 videos roughly. App is unusable.

**Current Workaround:** 

Current work around for me is to downgrade all the way back to 1.122.3 and disable auto-update in Play Store (ref: https://github.com/immich-app/immich/issues/14841#issuecomment-2557995492)

After downgrading to 1.122.3 following was observed: 
- Once sync is completed app can be used without the debilitating lag
- Albums don't go into an insane sync loop and lock everything up every startup.
- Assets from phone gallery show up instantly in 1.122.3 once permission is granted in 1.132.3 they take for ever to show up
- There is something in 1.122.3 that just works, swiping down on photos the animations work well too on 1.132.3
- I notice in 1.122.3 is that it caches more aggressively where as 1.132.3 barely passes 50MB. I personally prefer aggressive caching but then I have 512GB of storage.

I will be sticking to 1.122.3 until further notice as it's relatively feature complete for my use case.

---
P.S: Thank you for the brilliant project, and giving freedom from big brother. Looking to purchase it in full soon 💖

**Comment by akay64** [comment]
Current work around for me is to downgrade all the way back to [1.122.3](https://github.com/immich-app/immich/releases/tag/v1.122.3) and disable auto-update in Play Store (ref: https://github.com/immich-app/immich/issues/14841#issuecomment-2557995492)

After downgrading to 1.122.3 following was observed: 
- Once sync is completed app can be used without the debilitating lag unless something major changes in the library
- Albums don't go into an insane sync loop and lock everything up every startup.
- Assets from phone gallery show up instantly in 1.122.3 once permission is granted in 1.132.3 they take for ever to show up
- There is something in 1.122.3 that just works, swiping down on photos the animations work well too on 1.132.3
- I notice in 1.122.3 is that it caches more aggressively where as 1.132.3 barely passes 50MB. I personally prefer aggressive caching but then I have 512GB of storage. This could be placebo, but it's what I notice.

I will be sticking to 1.122.3 until further notice as it's relatively feature complete for my use case.

---
P.S: Thank you for the brilliant project, and giving freedom from big brother. Looking to purchase it in full soon 💖

**Comment by bo0tzz** [comment]
It's not resolved, as it says I closed it as a duplicate.

**Comment by Tronix117** [comment]
It seems a lot of duplicates has been closed but some are not duplicates actually... 
The freeze and lags due to sync on Android are pretty specific I guess (but I may be wrong). It also renders the app unusable on pretty much all Android phones (at least Samsung) without any way to make it work.
This issue contains a lot of infos compared to other issues
